EventStream/data/pytorch_dataset.py

```
@@ -126,137 +129,246 @@ def normalize_task(cls, col: pl.Expr, dtype: pl.DataType) -> tuple[str, pl.Expr]
+                    "start_time",
+                    "static_indices",
+                    "static_measurement_indices",
+                    "time_delta",
+                ],
+                use_pyarrow=True,
+            )
+
+            self.static_dfs[shard] = df
+            subject_ids = df["subject_id"]
+            n_events = df.select(pl.col("time_delta").list.lengths().alias("n_events")).get_column("n_events")
+            for i, (subj, n_events) in enumerate(zip(subject_ids, n_events)):
```

Ruff (0.8.2)

Loop control variable `n_events` overrides iterable it iterates

(B020)